.github/workflows/pages.yml

```diff
@@ -0,0 +1,56 @@
+name: Pages
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/configure-pages@v5
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: npm
+          cache-dependency-path: prototypes/basic/package-lock.json
+      - name: Setup hq
+        run: |
+          curl -sL https://github.com/ludovicianul/hq/releases/download/hq-1.3.2/hq_linux_amd64_1.3.2.tar.gz --output - | tar -xzv
+      - name: Build site
+        run: |
+          pushd ./lib/importer
+          npm install
+          popd
+
+          pushd ./prototypes/basic
+          npm install
+          coproc npm run dev &
+          while ! (echo >/dev/tcp/localhost/3000); do sleep 1; done
+          popd
+
+          pushd ./doc
+          export HQ=$(realpath ../hq)
+          make all
+          make all
```

Comment:
For some reason two of the assets don't get generated on the first run! I'm really pissed off and have no idea why, because avoiding that is like the whole point of make. But I've put more time into trying to work out why than I think its worth, so have settled for just running make twice. Sigh. Answers gladly received...

Reply:
Absolutely no idea unless it's the npm run dev taking its time to start up but presumably the `while` is waiting for that to be complete.